### ISSUE TITLE
Wireframe looking artefacts in 1.1.0

### ISSUE THREAD

**Issue by NevarcBB** [Mr-Wiseguy/Zelda64Recomp#289]
**DEV EDIT:**
If you are experiencing this glitch, the workaround for now is to go to `%LOCALAPPDATA%\Zelda64Recompiled` and in the `graphics.json` file change `api_option` to `Vulkan`. Another option is to change `hpfb_option` to `Off`, but doing so will cause the motion blur effect to not render correctly at high framerates. A proper fix or workaround will come later.

If you are experiencing the bug, we kindly ask you to make a comment in this issue with your GPU model and your driver version and **report the issue to AMD**. Thanks.

**Original post:**

Running the latest version on my system has a bunch of thin straight wireframe-esque lines around a bunch of objects in the game, including text boxes. Tested back and forth with 1.0.1 and it only happens on 1.1.0. Have also tested clearing files from appdata and clearing the driver cache, neither of which solved the issue.

Most of these screenshots are at 2x native due to the issue being more visible at lower resolutions. One native resolution (2560x1440) image is also attached to demonstrate what it looks like at high resolution.

![game1](https://github.com/Zelda64Recomp/Zelda64Recomp/assets/49603555/e5244d98-6de1-4f38-979b-95fe2a8f477e)
![game2](https://github.com/Zelda64Recomp/Zelda64Recomp/assets/49603555/211e53ff-a059-436f-93e1-fa33e64cc725)
![game3](https://github.com/Zelda64Recomp/Zelda64Recomp/assets/49603555/bdf1582a-2a92-45f0-8e02-650927cb51c2)
![game4](https://github.com/Zelda64Recomp/Zelda64Recomp/assets/49603555/ca92cb80-0d90-4094-a419-cda57c87162b)
![gamehires](https://github.com/Zelda64Recomp/Zelda64Recomp/assets/49603555/f3bef0ab-afeb-4ba2-b732-a5b09ed2e74d)
![menu](https://github.com/Zelda64Recomp/Zelda64Recomp/assets/49603555/13739f42-f5f2-4539-a25a-68f323d17d92)

 - OS: Windows 10 (10.0.19045 build 19045)
 - Version: 1.1.0
 - CPU: AMD Ryzen 5700X3D
 - GPU: Radeon RX 7900 XT

Tested on latest AMD driver at the time of writing (24.5.1)

**Comment by Mr-Wiseguy** [comment]
Thanks for the report. If you force use of the Vulkan API by editing the graphics.json config file and changing "api_option" to "Vulkan", do these problems still happen?

**Comment by Mr-Wiseguy** [comment]
Also, could you try changing the "hpfb_option" field from "Auto" to "Off" in that same config file?

**Comment by NevarcBB** [comment]
Yep just tested both options individually (forced Vulkan first, then set api back to Auto and tested hpfb_option being set to off) and they both result in the issue not appearing.

**Comment by Mr-Wiseguy** [comment]
Thanks for testing, it sounds like this may be a driver bug with HDR color formats in RDNA3 with D3D12, as we've tested RDNA2 on the latest drivers and didn't come across the issue. For now those two options are fine workarounds while we see if there's a workaround that can be done on the renderer side. If you're on a high refresh rate monitor then I suggest using the Vulkan workaround, but otherwise I'd use the hpfb_option workaround as D3D12 generally has lower input lag than Vulkan from our testing.

**Comment by NevarcBB** [comment]
Thanks for the help! I'm already running the game at 170Hz, so setting it to Vulkan is a fine solution for me. Hope you find a workaround though, and happy to volunteer my system to test any workarounds or driver updates.

**Comment by ripnetuk** [comment]
Can add to the weight of evidence that it's AMD related. Getting the same issue on a stock Lenovo legion go (ryzan z1 extreme integrated gfx/windows 11), and pleased to see workarounds . As an aside the free camera is beyond awesome :)
edit: Requested more specs for folk having this issue:

Graphics driver: Advanced Micro Devices, Inc.
07/11/2023
V 31.0.22024.3002

I think (but am not certain) that this is a Lenovo produced driver, and there is a beta driver out and about to be released soon. AIUI AMD license the Ryzon Z Extreme more cheaply if the OEM handles the driver/software side of things

CPU: AMZ Ryzen Z1 Extreme

OS - Windows 11 Pro 22631.3593

Machine itself - Lenovo Legion Go7, using the internal LCD

All Z64Recompiled settings as per default, except enabling the free camera, but also noticed without free camera enabled.
Thank you.

**Comment by DarioSamo** [comment]
Have any of you confirmed if it happens without MSAA?

I'm leaning very heavily towards this being a driver bug, so the expected workaround will likely be defaulting to Vulkan on the affected hardware until AMD looks into the issue.

**Comment by DarioSamo** [comment]
Apparently the same bug happens in RyujiNX in RDNA3:

![image](https://github.com/Zelda64Recomp/Zelda64Recomp/assets/538504/5e5c4a98-780d-49c2-92f5-9d18ff4f347b)

Seems we'll need a workaround for this family of GPUs.

**Comment by NevarcBB** [comment]
Just confirming that this issue happens regardless of MSAA setting. Initially tested this with MSAA off and have now tried with 2x and 4x. Only difference with the issue is that the lines do get anti-aliasing treatment.

**Comment by DarioSamo** [comment]
> Just confirming that this issue happens regardless of MSAA setting. Initially tested this with MSAA off and have now tried with 2x and 4x. Only difference with the issue is that the lines do get anti-aliasing treatment.

I'm very surprised it's not even related to the AA itself. This seems like the driver is fundamentally broken with this color format. I wonder if AMD might ship a hotfix for this soon. Thanks for the confirmation!

In the meantime I'll probably make it default to Vulkan on this family of GPUs if HDR support is requested. I don't see an alternative.

**Comment by YaSamu** [comment]
AMD Radeon RX 7800 XT 

AMD Windows Driver Version
31.0.24033.1003

Driver Version
23.40.33.01-240508a1-403196C-AMD-Software-Adrenalin-Edition

**Comment by febLey** [comment]
I can confirm this issue on an AMD Radeon 7800XT on the 24.4.1 driver.
Happens regardless of MSAA off/2x/4x setting.
Either changing the `api_option` to `Vulkan` or setting `hpfb_option` to `off`  in the `graphics.json` solved the issue.

**Comment by DarioSamo** [comment]
I'll be pushing a workaround for GPUs detected as "AMD Radeon 7". I'll post the PR for that here in a bit. Anyone with the `hpfb_option` should change it back to `Auto` and the API to D3D12 or Auto again and confirm if the issue goes away. The workaround simply will switch to Vulkan if it happens to detect the device name after creating it in D3D12. I am fairly certain this is a driver issue so I'm unable to work around it in any other way for now (unless someone else has an idea of what to attempt here, but without the hardware myself my options are limited).

I'm unable to find a way how to detect this on RDNA3 iGPUs. My own RDNA2 iGPU reports back as an unhelpful "AMD Radeon(TM) Graphic(s)" and I'm not sure what is reported back on an RDNA3 iGPU, and having to query all possible device Ids of these iGPUs is a bit beyond the scope of what I'd like to do here. I hope the bug gets fixed urgently.

**Comment by DarioSamo** [comment]
Get an artifact from this PR to test what I mentioned above: https://github.com/Zelda64Recomp/Zelda64Recomp/pull/314

**Comment by NevarcBB** [comment]
Yep, ran the artifact with renderer and hpfb set to Auto. No graphical issues, so I assume it has booted with Vulkan as expected.

**Comment by DarioSamo** [comment]
Thanks for the confirmation!

**Comment by nr1995** [comment]
Same problem here.
AMD 7800 X3D
AMD 7800 XT

**Comment by Standcat** [comment]
> Thanks for the report. If you force use of the Vulkan API by editing the graphics.json config file and changing "api_option" to "Vulkan", do these problems still happen?

??? Where the heck is this? I see no .json files anywhere. the only file named graphics i can find is a .rml...

**Comment by Maxmnf** [comment]
tengo un error lo abro y nisiquiera carga y c cierra solo y no me deja abrirlo ni nada
osea que abro el lanzador se queda la pantalla en negro y se cierra solo hay alguna solucion para eso?

**Comment by Reonu** [comment]
> tengo un error lo abro y nisiquiera carga y c cierra solo y no me deja abrirlo ni nada
> osea que abro el lanzador se queda la pantalla en negro y se cierra solo hay alguna solucion para eso?

Si tienes una gráfica Nvidia, actualiza los drivers a la última versión. Si eso no funciona, dinos las especificaciones de tu PC.

**Comment by Maxmnf** [comment]
> > tengo un error lo abro y nisiquiera carga y c cierra solo y no me deja abrirlo ni nada
> > osea que abro el lanzador se queda la pantalla en negro y se cierra solo hay alguna solucion para eso?
> 
> Si tienes una gráfica Nvidia, actualiza los drivers a la última versión. Si eso no funciona, dinos las especificaciones de tu PC.

las especificaciones de mi pc procesador :  sonntel(R) Core(TM) i3-4030U CPU @ 1.90GHz   1.90 GHz
ram instalada : 8,00 G
 que deberia de aser para solucionarlo ?

**Comment by Reonu** [comment]
> > > tengo un error lo abro y nisiquiera carga y c cierra solo y no me deja abrirlo ni nada
> > > osea que abro el lanzador se queda la pantalla en negro y se cierra solo hay alguna solucion para eso?
> > 
> > 
> > Si tienes una gráfica Nvidia, actualiza los drivers a la última versión. Si eso no funciona, dinos las especificaciones de tu PC.
> 
> las especificaciones de mi pc procesador : sonntel(R) Core(TM) i3-4030U CPU @ 1.90GHz 1.90 GHz ram instalada : 8,00 G que deberia de aser para solucionarlo ?

¿y la tarjeta gráfica?

como no has dicho tarjeta gráfica, entiendo que estás usando la integrada de ese procesador, es decir, una Intel HD Graphics 4400. Esa tarjeta gráfica no soporta DX12 ni Vulkan así que no puedes jugar. 

Los requisitos mínimos están en el readme, y tu ordenador no los cumple. Lo siento.

Además este hilo no tiene nada que ver con tu problema así que no pongas nada más al respecto en este hilo porque a la gente le van a llegar notificaciones de algo que no tiene nada que ver. Gracias

**Comment by rgraphic** [comment]
I'm using the ROG Ally (handheld gaming pc) and experienced the wireframe issue seen here. Sorry, im not nearly as knowledgeable as others on this or anything PC. I tried the "Vulcan" option and it didn't fix the issue, but the "hpfd" option did resolve it. 

Also just want to say I am just blown away by how cool it is to play an N64 era game like this. It is just so impressive what was achieved. Thank you. I cant wait to see this with Wave Race!

**Comment by Reonu** [comment]
Did you actually type "Vulcan"? that's a typo, it's Vulkan, with a k. That would be with it doesn't work, if you typed "Vulcan". 

Thanks for the kind words btw, I'm sure wiseguy and darío appreciate it

**Comment by rgraphic** [comment]
> Did you actually type "Vulcan"? that's a typo, it's Vulkan, with a k. That would be with it doesn't work, if you typed "Vulcan".
> 
> Thanks for the kind words btw, I'm sure wiseguy and darío appreciate it

Oh yeah, sorry I did type “Vulkan” when testing. It was just an autocorrect when typing the post here. Very happy the other option worked out.

**Comment by ripnetuk** [comment]
> > Did you actually type "Vulcan"? that's a typo, it's Vulkan, with a k. That would be with it doesn't work, if you typed "Vulcan".
> > Thanks for the kind words btw, I'm sure wiseguy and darío appreciate it
> 
> Oh yeah, sorry I did type “Vulkan” when testing. It was just an autocorrect when typing the post here. Very happy the other option worked out.

Did you check that it had persisted? first time I tried it, I edited the file, with Z64R open, and when I shut down Z64R it got put back. Its possible I mistyped it, or its possible Z64R saves its config on close.

I'm running on a Legion Go, which is a very similar device to yours, and once I closed the game, copied and pasted Vulkan into the file, saved, and re-launched the game, it fixed it perfectly.

**Comment by DarioSamo** [comment]
Has anyone been able to verify if the issue still persists on the latest driver? You'll likely need to force the configuration to "D3D12" for it to override the auto-switching.

**Comment by ripnetuk** [comment]
> Has anyone been able to verify if the issue still persists on the latest driver? You'll likely need to force the configuration to "D3D12" for it to override the auto-switching.

Hi. I can test it on the Lenovo legion Go if you like, but that device has it's drivers manager by Lenovo, so we get AMD driver updates after folk using the AMD provided drivers, and I'm not sure how the version numbers correlate.

Having said that, we have had one driver update since this issue was noticed.

I'll test it later and post back.

**Comment by ripnetuk** [comment]
@DarioSamo 
Hi, ive tested it on the latest drivers that Lenovo provide (they have some kind of odd agreement with AMD that they get the Z1 extreme chip cheaper if they handle the driver update side of things, so we are not using a vanilla AMD driver here)

The device is a Lenovo Legion Go with a Z1 Extreme APU

The problem remains.

![20240718_130946](https://github.com/user-attachments/assets/f0aa9201-a445-45ca-b6fb-0fa1e9cc1429)
![20240718_131012](https://github.com/user-attachments/assets/86b4d815-0265-4e86-822b-6fc3d5d369ac)
![20240718_131124](https://github.com/user-attachments/assets/49709d93-7b10-4f8b-9174-f70e705ed3c0)
